natlib/svcd.c

```diff
@@ -185,3 +186,41 @@ static int svcd_init( lua_State *L )
 
     return 0;
 }
+
+
+static int svcd_wcdispatch( lua_State *L )
+{
+    if (lua_gettop(L) != 3) return luaL_error(L, "Expected (pay, srcip, srcport)");
+    size_t parlen = lua_objlen(L, -1);
+    const char* pay = lua_tolstring(L, -1, &parlen);
```

Comment:
Thanks for your comments. We mixed up addressing on the stack. We need to negate all our addressing.